pkg/cmd/root.go

```diff
@@ -93,6 +97,10 @@ func kamelPreAddCommandInit(options *RootCmdOptions) *cobra.Command {
 }
 
 func kamelPostAddCommandInit(cmd *cobra.Command) error {
+	// Requires synchronization as viper bind flag is not able to handle concurrency
+	m.Lock()
+	defer m.Unlock()
+
```

Comment:
Wonder if the culprit is the fact that we are using a shared viper instance instead of using one per command instance

Reply:
I would not put too much effort into fixing this on the kamel CLI as it may be fixed already when using the Camel K JBang plugin instead

Reply:
100% agree, it was more a consideration about the original design

Reply:
@lburgazzoli BTW I ended up fixing the shared Viper instance so each command uses its own instance. Works like a charm now without the need for the lock on the mutex.

Reply:
very good, you cna probably blame me for that original design :)

Reply:
no offense to you the viper library totally offers the shared usage as a valid option. it took me some time to really understand why the tests hit the concurrency error though 😄